src/include/libpbs.h

```diff
@@ -138,6 +138,7 @@ pthread_mutex_t * get_conn_mutex(int);
 #define SVR_CONN_STATE_DOWN 0
 #define SVR_CONN_STATE_UP 1
 
+#define NSVR pbs_conf.pbs_num_servers
```

Comment:
I don't understand this. You are defining a pre-processor symbol as a variable name? Why?

Reply:
This saves the need of storing the variable in a temporary variable for iterating. See eefd054884c68a853edac96874b5b5c2d6229da6

Reply:
why not just use pbs_conf.pbs_num_servers directly then?

Reply:
that is a lengthy one and NSVR is used very often within for loop and such...

Reply:
As such, all of us use IDEs, I don't think long variable names are an issue, we have much longer pbs_conf variable names than this. But if you do want to shorten it, then just rename pbs_conf.pbs_num_servers instead of creating a level of indirection.

Reply:
I would prefer to keep NSVR if there are no issues with it.

Reply:
As such, all of us use IDEs, I don't think long variable names are an issue, we have much longer pbs_conf variable names than this. But if you do want to shorten it, then just rename pbs_conf.pbs_num_servers instead of creating a level of indirection, or use a local variable like before.

Reply:
There will not be an indirection as it is a macro.
The purpose of this change was to avoid creating a temporary variable every time you need to access frequently used variable like nsvr.

Reply:
what's wrong with a temporary variable? by indirection i was not talking about performance, i was talking about readability. This is totally unnecessary.

Reply:
The earlier temporary variable used was nsvr
That will now will be NSVR. which conveys the same meaning. How does it affect readability?

Reply:
you are creating a global reference to another global object just because you think it's too long, it will affect readability because it's not obvious why it's needed. Both of your reviewers are saying that they'd rather see pbs_conf.pbs_num_servers instead of NSVR everywhere, doesn't that tell you that your change might be affecting readability?

Reply:
Let me try to explain:
This is one of the instances where we have to replace NSVR with pbs_conf.pbs_num_servers
`for (i = start, ct = 0; ct < NSVR; i = (i + 1) % NSVR, ct++) {`
=>
`for (i = start, ct = 0; ct < pbs_conf.pbs_num_servers; i = (i + 1) % pbs_conf.pbs_num_servers, ct++) {`
This is the readability issue this change is trying to address. This is not an isolated occurrence. Most of the lines changed for NSVR are similar.

Accessing data using a getter function is a good practice. If such a function has to be used within a loop and such and does not want the overhead of function invocation, then it has to be treated as an inline function or declare as a macro. This what is done here.

Reply:
if you wanted to replace pbs_conf.pbs_num_servers with a getter then you should have used get_num_servers(). NSVR is not a getter, it won't help when we move to C++ as we'll need a proper function, not macro.

It sounds like the only "readability issue" that you are trying to solve is that "pbs_conf.pbs_num_servers" is too long. So, I again suggest to either rename it to make it shorter, or use get_num_servers(). If you still disagree, can we take this up later? The fixes that this PR adds are urgent, we are spending a lot of time discussing refactoring that's not needed for those fixes.

Reply:
get_num_servers() was the one that we were using initially. We started using a temp variable as most of the usage of this variable falls within loops and to avoid the function invocation overhead. Any names which you can form by reducing pbs_conf.pbs_num_servers will be still long for its use within a loop. I can declare a new global called nsvr and assign the value to that. But the present approach is even simpler.

It addresses the following:
+ Smaller name
+ Better readability
+ Does not have to use a temp variable.

If you think it won't help in the future, let's take it up at that point. what say?

Reply:
Sorry Nithin, I don't like this change, I don't think get_num_servers() will add any perceivable overhead (you are welcome to prove me wrong), I don't think the name is too long, and if it is then I don't see any issue with using local variables.

So, I'll again say, let's do this refactoring separately and we can involve more people and take their opinion. Since the fixes are urgent, I don't want to involve more folks as it'll delay the check-in.

Reply:
> I don't like this change

I read the whole thread again, the only point against the proposed change was that it adds a level of indirection. This is the case even if I revert back.

Reply:
Ok, I guess we are at a stalemate, I request opinions from other @openpbs/openpbs-maintainers on Nithin's change. Once we have more opinions and reach an agreement, we can proceed with the check-in.

Reply:
OK, since @agrawalravi90 wants our opinion, I'll give one.  I actually agree with Ravi.  NSVR isn't a getter.  If at some point in the future we want to replace it with a real getter, then we'll have to rename it to be a function (NSVR->get_nsvr()).  I don't mind the use of a temp variable because the compiler will optimize it out.  It'll see that the only use is pbs_conf.pbs_num_servers and either optimize both into a register, or just optimize it out and use the pbs_conf address itself.  I'd vote for a combination of the use of a temp variable or not.  Use the temp variable when the line gets too long, and don't use one if it isn't too long.